### PR TITLE
FEAT Rename a board

### PR DESCRIPTION
**Pull request by KenAlin**
Allows you to rename a board, by clicking on its title.

:information_source: Some changes where made to EditField.
* dragHandleProps : Is now not required
* EditField will now update state when props change. It is mandatory here, because we receive the board's title after EditField is rendered once.
* These changes doesn't break others EditField (for cards and lists titles)

### CODE REVIEW

File: src/commons/editField/EditField.jsx [hunk 1]
```diff
@@ -61,7 +65,7 @@ class EditField extends Component {
 EditField.propTypes = {
   text: PropTypes.string.isRequired,
   save: PropTypes.func.isRequired,
-  dragHandleProps: PropTypes.object.isRequired,
+  dragHandleProps: PropTypes.object,
```
Comment (Tombean) [inline src/commons/editField/EditField.jsx:78]: Remove it

**Review by Tombean — CHANGES_REQUESTED** [review]

File: src/containers/board/reducer.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { CLEAN_STATE, GET_ALL_LISTS_IN_BOARD, GET_ONE_BOARD } from './constants'
 
 const initialState = {
-  title: '',
+  data: [],
```
Comment (TheoGauchoux) [inline src/containers/board/reducer.js]: Why did you replace the `title` property by an array called `data` ?

File: src/commons/editField/EditField.jsx (hunk 1 above)
Comment (TheoGauchoux) [reply src/commons/editField/EditField.jsx:78]: No, I think we need it for card to just grab the title to drag and drop the card.

**Review by TheoGauchoux — DISMISSED** [review]

File: src/containers/board/reducer.js (hunk 2 above)
Comment (KenAlin) [reply src/containers/board/reducer.js]: To be more consistent with others reducers

File: src/commons/editField/EditField.jsx (hunk 1 above)
Comment (KenAlin) [reply src/commons/editField/EditField.jsx:78]: `.isRequired` is already removed

File: src/containers/board/reducer.js (hunk 2 above)
Comment (KenAlin) [reply src/containers/board/reducer.js]: Request for comment : data may be needed in a near future, to handle more properties than just "title" (private board, etc). So, do I have to keep "title" or should I change to "data" ?

Comment (TheoGauchoux) [reply src/containers/board/reducer.js]: The field `data` is used for collection reducers (like boardS, listS, cardS), not for board or card. `Data` is usually an array of element (of cards, board, lists, etc).

For an element container (boardContainer or cardContainer for example), you can flat the values directly in the state.

**Review by Tombean — APPROVED** [review]

**Pull request merged**